core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -1084,14 +1106,29 @@ object GroupMetadataManager {
    * @return payload for offset commit message
    */
   private[group] def offsetCommitValue(offsetAndMetadata: OffsetAndMetadata): Array[Byte] = {
-    // generate commit value with schema version 1
-    val value = new Struct(CURRENT_OFFSET_VALUE_SCHEMA)
-    value.set(OFFSET_VALUE_OFFSET_FIELD_V1, offsetAndMetadata.offset)
-    value.set(OFFSET_VALUE_METADATA_FIELD_V1, offsetAndMetadata.metadata)
-    value.set(OFFSET_VALUE_COMMIT_TIMESTAMP_FIELD_V1, offsetAndMetadata.commitTimestamp)
-    value.set(OFFSET_VALUE_EXPIRE_TIMESTAMP_FIELD_V1, offsetAndMetadata.expireTimestamp)
+    // generate commit value according to schema version
+
+    val offsetValueSchemaVersion: Short =
+      // if expire timestamp is explicitly provided use the old schema
+      if (offsetAndMetadata.expireTimestamp.nonEmpty) 1
+      //  otherwise, use the current schema
+      else CURRENT_OFFSET_VALUE_SCHEMA_VERSION
+
+    val value = new Struct(schemaForOffset(offsetValueSchemaVersion))
+
+    if (offsetValueSchemaVersion == CURRENT_OFFSET_VALUE_SCHEMA_VERSION) {
+      value.set(OFFSET_VALUE_OFFSET_FIELD_V2, offsetAndMetadata.offset)
```

Comment:
I may have had a comment about this already, but isn't it weird to assume v2 if the schema version matches `CURRENT_OFFSET_VALUE_SCHEMA_VERSION`? Perhaps we should get rid of `CURRENT_OFFSET_VALUE_SCHEMA_VERSION` and use v2 directly?

Reply:
Yeah I get the intent, but our execution was not great. Even with the update, we use `CURRENT_OFFSET_VALUE_SCHEMA_VERSION` initially, but then go on and assume that it must be version 2. We don't update this schema often, so I guess I'd rather just make it explicit.